### PR TITLE
INFRA-1673 ecs diskspace collector

### PR DESCRIPTION
**Pull request by burnsed**
Tested on our ecs dev machines. Will package this into a diamond-ecs container once merged (also already tested).

### CODE REVIEW

File: src/collectors/ecs_space/ecs_space.py [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+            return None
+
+        root_stats = os.statvfs(root_dir)
+        t = root_stats.f_bsize * root_stats.f_blocks
+        f = root_stats.f_bsize * root_stats.f_bfree
+
+        total = float(t) / 1024 / 1024 / 1024
+        available = float(f) / 1024 / 1024 / 1024
+        used = float(t-f) / 1024 / 1024 / 1024
+        utilization = used / (used + available) * 100
```
Comment (xavi-) [inline src/collectors/ecs_space/ecs_space.py]: Nit. `used / total`?

File: src/collectors/ecs_space/ecs_space.py [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+            raise e
+        except Exception as e:
+            self.log.error("Couldn't collect from docker: %s", e)
+            return
+
+        if available and used and total:
+            try:
+                available = self.parseHumanSize(available)
+                used = self.parseHumanSize(used)
+                total = self.parseHumanSize(total)
```
Comment (xavi-) [inline src/collectors/ecs_space/ecs_space.py:123]: This is a little sketchy.  Maybe unit tests for `parseHumanSize`?

File: src/collectors/ecs_space/ecs_space.py [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+            except ValueError as e:
+                self.log.error(str(e))
+                return
+
+            available = float(available) / 1000 / 1000 / 1000
+            total = float(total) / 1000 / 1000 / 1000
+            used = float(used) / 1000 / 1000 / 1000
```
Comment (xavi-) [inline src/collectors/ecs_space/ecs_space.py]: Might be slightly more readable if instead of `/ 1000 / 1000 / 1000` it used `/ UNIT_MAP['g']`?

**Comment by xavi-** [comment]
Other than a couple of nits, lgtm.

Also I don't quite understand how the unit tests work.  A lot of dynamism going on.

File: src/collectors/ecs_space/ecs_space.py (hunk 2 above)
Comment (burnsed) [inline src/collectors/ecs_space/ecs_space.py:123]: added a test for GB and MB. Note this is also mostly copied from https://github.com/docker/go-units/blob/master/size.go

**Comment by burnsed** [comment]
added more tests, incorporated @xavi-'s great suggestions and added some comments in the tests to explain them.

**Pull request merged**